pkg/ldapdn/ldapdn.go

```diff
@@ -0,0 +1,36 @@
+package ldapdn
+
+import (
+	"github.com/go-ldap/ldap/v3"
+	"golang.org/x/text/cases"
+)
+
+// While formally some RDN attributes could be casesensitive
+// maybe we should just skip the DN parsing and just casefold
+// the entire DN string?
+func NormalizeParsedDN(dn *ldap.DN) string {
+	var nDN string
+	caseFold := cases.Fold()
+	for r, rdn := range dn.RDNs {
+		// FIXME to really normalize multivalued RDNs we'd need
+		// to normalize the order of Attributes here as well
+		for a, ava := range rdn.Attributes {
+			if a > 0 {
+				// This is a multivalued RDN.
+				nDN += "+"
+			} else if r > 0 {
+				nDN += ","
+			}
+			nDN = nDN + caseFold.String(ava.Type) + "=" + caseFold.String(ava.Value)
+		}
+	}
+	return nDN
+}
+
+func NormalizeDN(dn string) (string, error) {
```

Comment:
Similarly to avoid stuttering and to improve the API signature of this module i sugget to name this function `ParseNormalize`.

Reply:
Makes sense. Updated.